crypto/src/key/hdkd/derivation_path.rs

```diff
@@ -54,6 +54,23 @@ impl DerivationPath {
     pub fn is_root(&self) -> bool {
         self.is_empty()
     }
+
+    /// Get the difference of this path and a sub path.
+    pub fn get_super_path_diff(&self, sub_path: &DerivationPath) -> Option<&[ChildNumber]> {
+        let super_path_vec = self.as_vec();
```

Comment:
Not related to this PR, but I just noticed this now. Isn't it better to make the `as_vec` method an `as_slice` and return `&[ChildNumber]` instead?